### ISSUE TITLE
Let the app load its intents/slots/… from a file

### ISSUE THREAD

**Issue by koenvervloesem** [rhasspy/rhasspy-hermes-app#11]
Let the app load its intents/slots/… from a file and re-train Rhasspy on installation/startup of the app.

See https://github.com/rhasspy/rhasspy-hermes/issues/12

**Comment by maxbachmann** [comment]
One thing to keep in mind with this, is that it should only retrain when there was a change (either to the files or some other settings like e.g. the language) and not on each restart, since training can take very long when a lot of sentences need to be generated.

**Comment by koenvervloesem** [comment]
Good point. So we should compare the locally defined intents and slots with the ones known/trained by Rhasspy then, and only trigger Rhasspy to re-train when they differ?

**Comment by maxbachmann** [comment]
I think it would be hard to compare them to the trained ones, since this would require generating them aswell.
Reasons to retrain from the top of my head:
- rhasspy settings change that require a retrain
- skill files change
- probably when the skill has a new version aswell, since following something like #13 e.g. the skill name might have changed

One more thing is that when skills retrain by sending their intents + slots we do want them to do this at the same time. It would be a complete mess when:
- skill A sends the intents
-> rhasspy starts to retrain
- skill B sends the intents
-> rhasspy has to retrain again

I am currently not sure how to implement this in an optimal way.
